src/Tonic/Generator.h

```diff
@@ -76,6 +83,17 @@ namespace Tonic {
       obj->tick(frames, context);
     }
 
+
+  //! Shortcut for wrapping with a Panner. Currently a MonoToStereoPanner but this could/should
+  // be replaced with a better panner that can handle mono or stereo.
+  Generator pan(ControlGenerator pan);
+
+  //! Shortcut for wrapping with a Panner. Currently a MonoToStereoPanner but this could/should
+  // be replaced with a better panner that can handle mono or stereo.
+  Generator pan(float pan);
+
+  void ____recordAdditionAsParameter____(){ obj->hasBeenAdded = true; }
```

Comment:
You have a class `A` and you want to give `B` and all of `B`'s descendents access to `A`'s private members. Have `A` make `B` a `friend` and then add a `protected` helper in `B` for `B`'s descendents to use.